features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapdConfigBean.java

```diff
@@ -82,4 +89,11 @@ public boolean getNewSuspectOnTrap() {
 		return m_snmpV3Users;
 	}
 
+	@Override
+	public void onUpdate(TrapdConfiguration config) {
+		this.m_snmpTrapAddress = config.getSnmpTrapAddress();
+		this.m_snmpTrapPort =config.getSnmpTrapPort();
+		this.m_snmpV3Users=new ArrayList<SnmpV3User>(TrapReceiverImpl.addToSnmpV3Users(config).values());
```

Comment:
m_snmpV3Users needs to be protected with some kind of synchronization so that it is thread-safe. It's probably appropriate to use Collections.synchronizedList() and synchronized{} blocks to protect it since it won't be accessed frequently.